### DISCUSSION TITLE
Best pipeline/practices for compressing scanned line art information using imagemagick convert

### DISCUSSION THREAD

**Discussion by zephyr707** [ImageMagick/ImageMagick#2979] [Help]
hello,

I am trying to determine the best pipeline for scanning and converting some documents (mostly lineart). The end product are rasterized non-vector pdfs. I've found that I need to scan at 600dpi+ to get acceptable results with line art settings and now I'd like to resize/resample/compress to reduce the file size.

What is the best scanning format to use, ppm, png, jpg, pdf? I have some old files scanned straight into pdf and have used imagemagick convert to adjust bit depth and resize them back to pdf, but I'm finding my repo's latest version on linux (Version: ImageMagick 6.9.10-23 Q16 x86_64 20190101) isn't responding the same way with my previous commands.

So now I've used pdfimages to extract to ppm, png, and tif and then used the following command:

`convert test-000.png -alpha off -monochrome -compress Group4 -quality 100 -units PixelsPerInch -density 600 test.pdf`

and that preserves the dimensions, but reduces the bit depth nicely and compresses things quite a bit, but is quite slow and only uses a single core. Is there a way to speed this up and use all the machine's resources? If I want to resize/resample should I chain another convert after this or is there a way to do it in one command without sacrificing quality?

any advice would be much appreciated, thanks!

**Comment by snibgo** [comment]
It depends on what the scans are for. If this is to keep a record (but not OCR) of business lunch receipts etc, your scheme seems okay.

But for digital versions of old documents, I would never suggest reducing the images to 1 bit per pixel, before compression. This is really very low quality.

If you want to save at different sizes, you can do that in one command. Ensure you are making a resize of the original, not of a compressed version.

**Reply by zephyr707** [reply]
thanks for your reply, snibgo, this is exactly the kind of guidelines I'm looking for. These will be for archival purposes, but won't necessarily ever be printed again, so really as long as they are readable on a screen that's great. If I still want them to be ready for OCR what are the recommended settings/guidelines? Will 1-bit depth kill any hopes of OCR? I haven't found a great solution for OCR on linux, but would like to reserve the ability to create searchable documents at some point.

I did find that the -monochrome was the slow factor, (maybe due to dithering?), threshold 50% was much faster. I am resizing the png/ppm/tif and getting that to a state that seems acceptable before moving on to the conversion to pdf. I've also read that ghostscript may be a valuable tool when working with pdfs. The pdf format is being used simply because it seems widely compatible.

**Comment by snibgo** [comment]
For accurate OCR using Tesseract, I suggest the height of characters such as "M" should be at least 20 pixels, and the image should be grayscale (at least 8 bits). If the depth is only 1 bit, I suppose a larger character size would compensate, eg a height of at least 40 pixels.

Your mileage may vary. I don't have much experience with OCR.

IM uses Ghostscript to rasterize PDFs. When writing PDFs, it simply creates a PDF wrapper around the raster images.

If you are writing each raster image to its own PDF, I can see no advantage to using PDF. It simply adds an extra layer of complexity.

**Comment by zephyr707** [comment]
Thanks for your advice, I will try to find a doc/source with regards to best practices for OCR, but if you have any handy please do share.

Ok, great to know that IM is using GS with PDFs. What format do you recommend instead of PDF? Maybe it was not clear in my first post, but most of these documents will be multi-paged documents. I believe tiff can do those as well although I've never done it. Pdf has always seemed the default for me since it is so widely used and accepted by other programs and if I plan on adding OCR/text searchability it seems to be a future proof choice along with the other features Pdf brings.

I'm mostly going off of information in docs such as: [this](https://www.foxitsoftware.com/blog/pdf-vs-tiff-which-is-the-right-format-for-your-document-scanning-needs/) and [this](https://www.aquaforest.com/en/tiff_versus_pdf.asp), but if there is some other format I should be considering please point me in the right direction, thanks!

If I stay with a final pdf format it seems like scanning in png or tif and then using IM to resize/adjust before converting to pdf. I do see a lot of OCR pre-processing threads on the old board index, so will have a look at those, too.

**Comment by snibgo** [comment]
A TIFF file can contain multiple images.

On PDF versus TIFF, I suggest that again this depends on the purpose. PDF support for accurate colour management seems to be weak, but this may not matter for you. PDF has more structured metadata for storing OCR text, so should be better for searching and indexing, but you are at the mercy of whatever tools you use for that.

A decision on format shouldn't restrict future choices. The nature of compression and bit depth does restrict future choices because once you have removed data, you can't get it back.
